src/components/BDetails/BDetails.vue

```diff
@@ -0,0 +1,45 @@
+<template>
+<details>
+  <summary>
+    <strong>{{summary}}</strong>
+    <div class="icon">
+      <slot name="icon"></slot>
+    </div>
+  </summary>
+  <slot></slot>
+</details>
+</template>
+
+<script>
+export default {
+  name: 'b-details',
+  props: {
+    summary: {
+      type: String,
+      required: true
+    }
+  }
+}
+</script>
+
+<style>
+.icon {
+  float: right;
+}
+details[open] summary ~ * {
+  animation: fadein 1s;
+}
+@keyframes fadein {
+  from { opacity: 0; }
+  to { opacity: 1; }
+}
+</style>
+
+<docs>
```

Comment:
I think we are going to be using .md files instead of doc tags

Reply:
K, this was the model I had found. Can change it.